eng/pipelines/runtime-extra-platforms.yml

```diff
@@ -541,8 +538,7 @@ jobs:
       condition: >-
         or(
           eq(dependencies.evaluate_paths.outputs['SetPathVars_wasmdebuggertests.containsChange'], true),
-          eq(variables['isManualOrIsNotPR'], true),
-          eq(variables['isFullMatrix'], true))
+          eq(variables['isRollingBuild'], true))
```

Comment:
This would mean that when `runtime-extra-platforms` is being run, then it will run only if the changes were found. But isn't the expectation that when running this pipeline manually, everything should run?
Some of the other jobs in this file also seem to have the same issue.

Reply:
Yes, I would expect at least the configurations that don't run full tests on PR's to run all when manually triggered.

Reply:
Since this is now not longer a mono specific pipeline like the runtime-manual was, we are still using the evaluate-paths logic for PRs but scheduled will run all. 

However, we can adjust these conditions as convenient, for example instead of depending on the wasm debugger path subset, we can just make this less granular and depend on any mono change.

Reply:
That kinda defeats the purpose of running it manually, IMHO. IIUC, for PRs we run a limited set by default, like smoke tests for libraries. And we had added this separate pipeline to allow running *all* the tests, so we could check in the PR if anything broke. Otherwise, we would find out only on a rolling/scheduled build.

Please correct me if I'm wrong.

Reply:
I think I'm not being clear. If you change a path that applies to any of the needed tests, it will run all the tests, but only if you changed paths that need those tests.

Reply:
Right, so if the path condition doesn't match the changes then it won't run that job. But in `runtime-manual` we were always running the jobs, so it covered cases that you can't easily cover through path conditions. Eg. when making changes in sendtohelix*, or some Directory.Build.props, which isn't directly under a path for wasm debugger tests, or wasm build tests.

Reply:
The big difference with runtime-manual is that this new pipeline now has way more tests than just mobile, so if we ran everything for a change that only modified say a runtime test, we will waste resources in a manner that is not useful. 

That said, if we care about running all mono tests when a Directory.Build.props is changed or when a sendtohelix is changed, we can change the debugger or wasm build test condition either to include more paths or tu just use the `mono` subset path which already includes those paths:

https://github.com/dotnet/runtime/blob/main/eng/pipelines/common/evaluate-default-paths.yml#L31-L50

Reply:
Can we split the wasm ones from this pipeline then? Mobile ones probably also want to run similarly. [Here](https://github.com/dotnet/runtime/issues/62276) are some examples of cases where issues got caught in the rolling build instead of the PR itself, thus affecting many more people than just the PR author.
@steveisok @lewing

>  so if we ran everything for a change that only modified say a runtime test, we will waste resources in a manner that is not useful

True, that's why it should happen only for rolling builds, or when run manually.

Reply:
> True, that's why it should happen only for rolling builds, or when run manually.

But we only have either rolling or manual trigger with this pipeline. We don't have per PR run by default, only if invoked manually. 

> Can we split the wasm ones from this pipeline then? Mobile ones probably also want to run similarly. Here are some examples of cases where issues got caught in the rolling build instead of the PR itself, thus affecting many more people than just the PR author.

We have been using the evaluate-changed-paths ever since the repo was created and we have found very fue issues because of that but a huge gain on resource saving. I don't think we should throw in and create a pipeline for mono and one for non mono, that will cause more and more confusion to developers. Running based on changed paths is the same as if it were part of the main PR pipeline, even the smoke tests run only if mono or libraries are changed.

Rather than throwing more pipelines into the equation I would just update the legs that depend on wasm debugger or build browser paths, to instead depend on mono paths + debugger paths/build browser paths.

Reply:
> But we only have either rolling or manual trigger with this pipeline. We don't have per PR run by default, only if invoked manually.

Oh, then it would mean that the browser/windows jobs will run only on rolling builds, or manual runs. Was that an oversight?

Reply:
> Yes, I would expect at least the configurations that don't run full tests on PR's to run all when manually triggered.

I missed we were setting evaluate paths. When triggered manually, we would be getting the same experience before we started running smoke tests only.  That should be acceptable for most cases.

For changes that cause no legs to run, I think we either do what santi suggested and adjust specific paths or there is always the option to trigger manually from azdo.  You can set any yml variable and force a full run if you wanted to.

Reply:
> Oh, then it would mean that the browser/windows jobs will run only on rolling builds, or manual runs. Was that an oversight?

They will run on manual runs or scheduled runs for the extra-platforms pipeline. If we want them to run smoke tests on the main runtime pipeline, then we can do that as well.

> For changes that cause no legs to run, I think we either do what santi suggested and adjust specific paths or there is always the option to trigger manually from azdo. You can set any yml variable and force a full run if you wanted to.

Yeah, however if you set the mono subsets path + the wasm debugger and the build browser paths, I think it will be very very few cases where you will need to run it manually from the pipeline UI

Reply:
> If we want them to run smoke tests on the main runtime pipeline, then we can do that as well

Yeah, runtime-staging. I can open a PR for that